src/memote/consistency.jl

```diff
@@ -321,3 +321,92 @@ function is_consistent(
     value.(m)
     termination_status(opt_model) == OPTIMAL
 end
+
+"""
+$(TYPEDSIGNATURES)
+
+Test if model is consistent by checking that:
+1. the model is stoichiometrically consistent, tested with
+   [`is_consistent`](@ref)
+2. there are no energy generating cycles, tested with
+   [`has_erroneous_energy_generating_cycles`](@ref)
+3. the model is both mass and charge balanced, tested with
+   [`reaction_charge_unbalanced`](@ref) and [`reaction_mass_unbalanced`]
+
+Each function called in this test function can be called individually. The
+kwargs are forwarded as indicated by the prefix.
+"""
+function test_consistency(
+    model,
+    optimizer;
+    consistency_boundary_reactions = [rid for rid in reactions(model) if is_boundary(model, rid)],
+    consistency_ignored_reactions = find_biomass_reaction_ids(model),
+    energy_dissipating_energy_dissipating_metabolites = Dict(
+        "ATP" => "atp_c",
+        "CTP" => "ctp_c",
+        "GTP" => "gtp_c",
+        "UTP" => "utp_c",
+        "ITP" => "itp_c",
+        "ADP" => "adp_c",
+        "CDP" => "cdp_c",
+        "GDP" => "gdp_c",
+        "UDP" => "udp_c",
+        "IDP" => "idp_c",
+        "NADH" => "nadh_c",
+        "NAD" => "nad_c",
+        "NADPH" => "nadph_c",
+        "NADP" => "nadp_c",
+        "FADH2" => "fadh2_c",
+        "FAD" => "fad_c",
+        "FMNH2" => "fmn_c",
+        "FMN" => "fmnh2_c",
+        "Ubiquinol-8" => "q8h2_c",
+        "Ubiquinone-8" => "q8_c",
+        "Menaquinol-8" => "mql8_c",
+        "Menaquinone-8" => "mqn8_c",
+        "2-Demethylmenaquinol-8" => "2dmmql8_c",
+        "2-Demethylmenaquinone-8" => "2dmmq8_c",
+        "ACCOA" => "accoa_c",
+        "COA" => "coa_c",
+        "L-Glutamate" => "glu__L_c",
+        "2-Oxoglutarate" => "akg_c",
+        "Ammonium" => "nh4_c",
+        "H" => "h_c",
+        "H[external]" => "h_e",
+        "H2O" => "h2o_c",
+        "Phosphate" => "pi_c",
+        "Acetate" => "ac_c",
+    ),
+    energy_dissipating_additional_energy_generating_reactions = [],
+    energy_dissipating_ignored_reactions = ["ATPM"],
+    energy_dissipating_modifications = [],
+    energy_dissipating_boundary_reactions = [rid for rid in reactions(model) if is_boundary(model, rid)],
+    mass_ignored_reactions = [
+        find_biomass_reaction_ids(model)
+        find_exchange_reaction_ids(model)
+    ],
+    charge_ignored_reactions = [
+        find_biomass_reaction_ids(model)
+        find_exchange_reaction_ids(model)
+    ],
+)
+    @testset "Consistency" begin 
+        @test is_consistent(
```

Comment:
function name: consistent in regard to what model?

Reply:
the model itself (it is what memote calls it)

Reply:
scary. Universal unified consistency. :D